nixos/modules/services/security/tor.nix

```diff
@@ -962,8 +965,8 @@ in
               '') onion.authorizedClients ++
               optional (onion.secretKey != null) ''
                 install -d -o tor -g tor -m 0700 ${escapeShellArg onion.path}
-                key="$(cut -f1 -d: ${escapeShellArg onion.secretKey})"
```

Comment:
Since the newline can only appear after the `== ed25519v1-secret` header, I think it's enough to just add `| head -1` after this `cut`.

Reply:
how about this solution (using `-s`) ?

Reply:
ah, probably not, since : can also appear after a newline in the binary I guess?